packages/x-date-pickers/src/internals/hooks/usePicker/usePickerValue.ts

```diff
@@ -429,7 +437,10 @@ export const usePickerValue = <
   const handleDismiss = useEventCallback(() => {
     // Set all dates in state to equal the last committed date.
     // e.g. Reset the state to the last committed value.
-    setDate({ value: dateState.committed, action: 'acceptAndClose' });
+    setDate({
+      value: resetValueOnClose ? dateState.resetFallback : dateState.committed,
+      action: 'acceptAndClose',
+    });
   });
```

Comment:
I don't get why the modification is here. I agree "dismiss" is mostly used when closing UI. But I doubt it's a good idea to have `onDismiss` behavior depending on `resetValueOnClose` especially if at some point we open access to hook. 

For me, it would make more sense that depending on `resetValueOnClose` we either call `onDismiss` or `onCancel`

Reply:
Very good point. 👌

Reply:
[Updated](https://github.com/mui/mui-x/pull/7721/commits/e6075da46b352fc125a1f8cad322a75c4dc48814)